### DISCUSSION TITLE
Importance of MPDist in the .snippets

### DISCUSSION THREAD

**Discussion by adityabhandwalkar** [TDAmeritrade/stumpy#813] [General]
Hello community , 

While working with `.snippets` of the stumpy which yields best subsequence representing the time series. I opted 2 best subsequence and tried to plot the MPdist plot as shown below using the  [tutorial](https://github.com/TDAmeritrade/stumpy/blob/main/docs/Tutorial_Time_Series_Snippets.ipynb)
![MPdistplot](https://user-images.githubusercontent.com/56731860/225033282-4b72e348-269b-4fb9-bc19-4c72b680af73.png)

My query is how MPDist  can provide useful insights about the the snippet. As MPdist tell us are two time series related to each other. and in `.mpdist` returns the smallest distance between two time series, so it means lower the value of Mpdist more likely two time series are similar? Is it correct to say. In addition to that lets say when I have very huge data set where i slice down the data in two halves and I get 5 subsequence each  in total 10 snippets using the .`snippet`  how can we compare these snippets with each other on basis of MPDist as stated in above figure I cannot understand how it would help me to determine the good snippets. Secondly if I were to classify these 10 snippet on basis of Mpdist  score e.g. lowest value snippet is good one and one with highest MPdist is bad snippet of time series 

Thank you

**Comment by adityabhandwalkar** [comment]
@seanlaw  do you have any suggestions in this context.

**Reply by seanlaw** [reply]
> As MPdist tell us are two time series related to each other. and in .mpdist returns the smallest distance between two time series, so it means lower the value of Mpdist more likely two time series are similar? 

Yes, I believe that is the correct interpretation. I recommend reading through the original [MPdist paper](https://ueaeprints.uea.ac.uk/id/eprint/69244/1/ICDM_2018_Shaghayegh.pdf) to get a firm grasp of how to interpret it.

> Is it correct to say. In addition to that lets say when I have very huge data set where i slice down the data in two halves and I get 5 subsequence each in total 10 snippets using the .snippet how can we compare these snippets with each other on basis of MPDist as stated in above figure I cannot understand how it would help me to determine the good snippets. Secondly if I were to classify these 10 snippet on basis of Mpdist score e.g. lowest value snippet is good one and one with highest MPdist is bad snippet of time series

I don't have much experience with snippets so I recommend going over the [original snippets paper](https://www.cs.ucr.edu/~eamonn/Time_Series_Snippets_10pages.pdf) and understanding its use from there. However, based on the numerous questions that you've asked recently, it seems that you are hopping from function to function without taking the time to educate yourself on their origins (i.e., by reading the original published papers) and how to interpret their results. In nearly all of our docstrings, there is a Notes section that provides a DOI reference that points back to the original published papers and I strongly urge you to go through them as we are ill-equipped to provide support on deep research questions as we likely don't have the context for your particular domain of research/work. Please understand that this is all volunteer work and STUMPY is really only here to faithfully reproduce the original published work. So, I also recommend emailing the original authors when you aren't able to get a proper response here. I can tell you that I would've responded to your question if I knew the answer but, sadly, I am unable to help.

Best of luck

**Reply by adityabhandwalkar** [reply]
Thanks for your reply and support , Just wanted to clear my doubt about it, Its not that I have not read the paper but sometimes interpretation of the result also plays an important role while working with any library. I did not directly hop into the function used it before that I  did read the paper and gone through theory stuff, but when I was unable to understand it, Please pardon me if I caused any trouble with this 

Thanks

**Reply by seanlaw** [reply]
@adityabhandwalkar No need to apologize. All of these publications are still new and it has yet to be revealed as to which functions are actually useful. Unfortunately, we are all working at the edge of our knowledge and understanding. In the future when posting a question, I recommend:

1. Clearly explain what the basic problem is that you are trying to solve. Your previous posts were very hard to follow due to grammatical errors and your intent/question was lost.
2. What other resources have you referenced (citing exact lines of code or text) so that others can understand your problem (most of your posts have been really hard to follow/understand as they lack context and a clear goal)
3. What have you tried and why didn't it work?
4. Provide a minimum reproducible example with code and data (i.e., don't just provide a plot that is only meaningful to you). When you fail to do this, you are making it the job of others to read your mind and nobody has the time or resources to help you.

All of this will improve the likelihood (but does no guarantee) that you will receive a response. I hope that helps
